### DISCUSSION TITLE
Desktop cannot launch applications after long periods of "idle"

### DISCUSSION THREAD

**Discussion by JamesShriner** [lxqt/lxqt#2489] [General]
##### Expected Behavior

Applications should launch without issue
##### Current Behavior

Applications do not launch.  Not from menu, not from lxqt-panel quick launchers, not from WM (fluxbox) root menu.  If an application is already open (such as qterminal), that application works; but no new instances.  attempting to launch applications from qterminal fails, but with error message:
Maximum number of clients reached
Error: cannot open display: :0
##### Possible Solution

This appears to be an issue with lxqt-config-monitor on a brand new installation to a refurbed Lenovo ThinkCentre m715q.  Brand new installations on other machines have no such issues.  Testing other LiveUSBs on the problem machine indicate the problem is limited to LXQT desktop.  Other Debian distros did not exhibit same behavior.
##### Steps to Reproduce (for bugs)

1. open qterminal (for troubleshooting after wake from idle)
2. Allow system to idle, and screen to darken
3. Wake system with mouse/keyboard event
4. attempt to launch new GUI application (menu, launcher, rootmenu.....all fail)
5. attempt to launch from terminal prompt (error message indicated above)
6. xrestop command indicates "monitoring 256 clients" (the maximum limit?) with NUMEROUS xerrors
7. lsof -U provides list of running files/processes on system
8. pkill "process name" (lxqt-conf according to lsof command)
9. xrestrop indicates "monitoring 40 clients" with NUMEROUS xerrors
10. application launch is restored (via all options, menu, launcher, root menu, and terminal)
11. allow system to idle and repeat ad nauseum

##### Context

This issue makes LXQT unusable on this machine....my PRIMARY machine.  It looks like a workaround is to increase the number of xclients allowed in xsession, but that seems like it will only "increase the headroom" and result in the same issue, but longer intervals
##### System Information

* Distribution & Version: Debian 12 Bookworm
* Kernel: 6.1.0-13
* Qt Version: 1.2
* liblxqt Version: ??  not sure where to find this
* Package version: ?? not sure where to find this
* Inxi -Fxz provides following (I suspect Ryzen5 RavenRidge/Vega hardware may be an issue here, and 2019 BIOS was updated to June 2023 BIOS, but with no change in behavor):
* System:
  Kernel: 6.1.0-13-amd64 arch: x86_64 bits: 64 compiler: gcc v: 12.2.0
    Desktop: LXQt v: 1.2.1 Distro: Debian GNU/Linux 12 (bookworm)
Machine:
  Type: Mini-pc System: LENOVO product: 10VHS17600 v: ThinkCentre M715q
    serial: <superuser required>
  Mobo: LENOVO model: 3130 v: SDK0J40697 WIN 3305183174933
    serial: <superuser required> UEFI: LENOVO v: M1XKT62A date: 06/21/2023
CPU:
  Info: quad core model: AMD Ryzen 5 PRO 2400GE w/ Radeon Vega Graphics
    bits: 64 type: MT MCP arch: Zen rev: 0 cache: L1: 384 KiB L2: 2 MiB
    L3: 4 MiB
  Speed (MHz): avg: 1574 high: 1600 min/max: 1600/3200 boost: enabled cores:
    1: 1538 2: 1600 3: 1454 4: 1600 5: 1600 6: 1600 7: 1600 8: 1600
    bogomips: 51104
  Flags: avx avx2 ht lm nx pae sse sse2 sse3 sse4_1 sse4_2 sse4a ssse3 svm
Graphics:
  Device-1: AMD Raven Ridge [Radeon Vega Series / Radeon Mobile Series]
    vendor: Lenovo driver: amdgpu v: kernel arch: GCN-5 bus-ID: 04:00.0
    temp: 48.0 C
  Display: x11 server: X.Org v: 1.21.1.7 driver: X: loaded: amdgpu
    unloaded: fbdev,modesetting,vesa dri: radeonsi gpu: amdgpu
    resolution: 1920x1080~60Hz
  API: OpenGL v: 4.6 Mesa 22.3.6 renderer: AMD Radeon Vega 11 Graphics
    (raven LLVM 15.0.6 DRM 3.49 6.1.0-13-amd64) direct-render: Yes
Audio:
  Device-1: AMD Raven/Raven2/Fenghuang HDMI/DP Audio driver: snd_hda_intel
    v: kernel bus-ID: 04:00.1
  Device-2: AMD ACP/ACP3X/ACP6x Audio Coprocessor driver: snd_pci_acp3x
    v: kernel bus-ID: 04:00.5
  Device-3: AMD Family 17h/19h HD Audio vendor: Lenovo driver: snd_hda_intel
    v: kernel bus-ID: 04:00.6
  API: ALSA v: k6.1.0-13-amd64 status: kernel-api
  Server-1: PulseAudio v: 16.1 status: active
Network:
  Device-1: Realtek RTL8111/8168/8411 PCI Express Gigabit Ethernet
    vendor: Lenovo driver: r8169 v: kernel port: f000 bus-ID: 01:00.0
  IF: eth0 state: up speed: 1000 Mbps duplex: full mac: <filter>
  Device-2: Intel Wireless-AC 9260 driver: iwlwifi v: kernel bus-ID: 02:00.0
  IF: wlan0 state: down mac: <filter>
Bluetooth:
  Device-1: Intel Wireless-AC 9260 Bluetooth Adapter type: USB driver: btusb
    v: 0.8 bus-ID: 3-1.4:3
  Report: hciconfig ID: hci0 rfk-id: 1 state: up address: <filter> bt-v: 3.0
    lmp-v: 5.1
Drives:
  Local Storage: total: 238.47 GiB used: 21.17 GiB (8.9%)
  ID-1: /dev/nvme0n1 vendor: Samsung model: MZVLB256HAHQ-000L7
    size: 238.47 GiB temp: 34.9 C
Partition:
  ID-1: / size: 233.38 GiB used: 21.15 GiB (9.1%) fs: ext4 dev: /dev/nvme0n1p2
  ID-2: /boot/efi size: 299.4 MiB used: 20 MiB (6.7%) fs: vfat
    dev: /dev/nvme0n1p1
Swap:
  ID-1: swap-1 type: file size: 512 MiB used: 0 KiB (0.0%) file: /swapfile
  ID-2: swap-2 type: zram size: 13.84 GiB used: 0 KiB (0.0%) dev: /dev/zram0
Sensors:
  System Temperatures: cpu: 47.6 C mobo: N/A gpu: amdgpu temp: 47.0 C
  Fan Speeds (RPM): N/A
Info:
  Processes: 269 Uptime: 11h 54m Memory: 14.57 GiB used: 2.21 GiB (15.2%)
  Init: systemd target: graphical (5) Compilers: gcc: 12.2.0 Packages: 2084
  Shell: Bash v: 5.2.15 inxi: 3.3.26

**Comment by stefonarch** [comment]
This is hardly an LXQt issue but (my guess) an environment one:

> Error: cannot open display: :0

What do you see with `env |grep -i display` in this condition?

Moving to "Discussions".

**Reply by JamesShriner** [reply]
thanks for such a QUICK response @stefonarch!  terminal results of env | grep -i display are:
XDG_SESSION_PATH=/org/freedesktop/DisplayManager/Session0
XDG_SEAT_PATH=/org/freedesktop/DisplayManager/Seat0
DISPLAY=:0

**Reply by JamesShriner** [reply]
Stefonarch...you have my attention with the "environment" issue.  Not sure what that means, but I think you're onto something...potentially something quick & easy, but too elusive for me.

I disconnected the "problem" machine from my monitor and slid a new machine in its place for the same setup, but using intel Core i7 processor.  This machine tested fine with the LiveMedia, but having display problems after installation to metal.

Meanwhile, I plugged an old 4:3 monitor into the "problem" machine with the Ryzen 5 Pro, just so I could perform whatever commands you suggest/recommend.  I shook the mouse today for wakeup, and this thing seems to be running fine, with no zombie xclient connections being spawned.  it continues to hover around 40 clients (similar to my Debian 10 machine.....a known good configuration that's never had issues).

The problem seems to travel to the Debian 12 machine hooked up to the primary monitor....a 1920x1080 27" AOC LCD.  My intel i core 7 machine has problems recognizing screen size upon wake, with Tint2 moving from the intended right edge (centered) to the top left corner.  I also note that an lxpanel gets slightly re-positioned too.  And some applications (notably firefox and qterminal) lose their window decorations until I restart my window manager (Fluxbox), and everything is fine again until the next idle/sleep session.  Seems like machines connected to this monitor have a hard time gleaning settings for Xorg or something.  But before you suspect the monitor, let me confirm that it has been working fine with NO ISSUES whatsoever in Debian 10 for about 3 years, and in Ubuntu 16.04 for about 3 years before that.  So I'm pretty confident the hardware is okay.

But back to the Ryzen 5 machine that I originally posted about.  With the information provided herein, is there anything I can do to make this 27" 1920x1080 monitor work with the Ryzen 5 machine?  While the 4:3 lcd is working fine, it's just too small for daily use.  It's only 15", and the 4:30 ratio just won't suit my work needs.  I really need the widescreen 16:9 monitor.

Any advice on how to make that monitor work for me?

**Comment by JamesShriner** [comment]
EDIT:  I can also report 1 other variable....the monitor connection.  The "problem monitor" is connected with HDMI to DisplayPort adaptor, into Display Port on machine.  The "good monitor" is connected with VGA cable to DisplayPort adaptor into Display Port on machine.

Although both machines require DisplayPort connections to monitor, neither monitors have DisplayPort output.  Would HDMI vs VGA be a potential culprit also?

**Reply by tsujan** [reply]
A quick google search showed this: https://unix.stackexchange.com/questions/368017/cant-start-applications-due-to-maximum-number-of-clients-reached-error

**Reply by JamesShriner** [reply]
thanks for the reply tsujan!  That is exactly my problem!  And that is the thread that helped me determine that lxqt-config-monitor was the zombie process spawning all my xclient connections.  Which is why I filed a bug report, but was converted to this discussion.  Stefon suggested an "environment" issue.  I've provided requested information above, but that's as far as we got.  

Based on today's events, I thought it might be helpful for Stefon (or whomever) to inform of the issue seeming to be monitor-related, rather than software-related.  Hoping the additional information confirms whatever he suspects?

I thought it was encouraging to see a different monitor provide different results.  Seems to be a configuration issue of some sort?  How can I get my BIG 16:9 monitor to behave like the little 4:30 monitor?  any advice?

**Reply by tsujan** [reply]
I don't have the required knowledge to answer your question. But if you think that lxqt-config-monitor is the culprit (you don't need to be sure), please open a report at https://github.com/lxqt/lxqt-config/issues, with the relevant info. Please make sure that the report is concise: fill in the issue template and provide all the info needed.

**Reply by JamesShriner** [reply]
thanks tsujan.  i did file bug report #2488, but it was immediately closed and moved to this discussion.  I'm not sure why it was closed, but I trust that LXQT devs would know.  Until I hear from someone, I'm at a loss for what to do next?  But I do appreciate that you tried to help me!  THANK YOU!

**Reply by tsujan** [reply]
It wasn't closed but converted to a discussion, and there was a good reason for that: users with similar experience might see it more easily here than there, while it didn't *seem* to be an LXQt issue.

Now, when you said, "lxqt-config-monitor was the zombie process spawning all my xclient connections," things seem different.

**Reply by JamesShriner** [reply]
aha....I understand.  Bug report #2488 was filed.  Hoping all the information provided is helpful.  Again, thanks!

**Reply by tsujan** [reply]
> Hoping all the information provided is helpful.

The problem was that the relevant info could hardly to be seen in it ;) It wasn't concise. Now we know that, in your case, lxqt-config-monitor doesn't work fine: for some reason it *exhausts* connections. That may be really a bug in lxqt-config-monitor, whose issue tracker is https://github.com/lxqt/lxqt-config/issues. It deserves a thorough investigation.

**Comment by JamesShriner** [comment]
Progress update....building on the unexpected surprise discovery of the VGA monitor with DP adaptor working just fine with the Ryzen machine, I found a recessed VGA connection in the "problem" monitor and booted into the OS with a nomodeset GRUB parameter (which wasn't required on the "good" VGA monitor).  Opened a terminal and ran xrestop to monitor xclient connections.  Let the machine idle/sleep and walked away long enough for xclient connections to accumulate.  30 minutes later, there were NO new connections.

In the interests of thoroughness, swapped the Intel back to the "problem" monitor with VGA/DP connection, and the quirky desktop behavior described above was gone.  I let the machine idle/sleep and woke it up after about 30 minutes....no xclient connections spawned (no different than before), but the tint window and lxqt panel remained where they were supposed to be.

This seems to suggest a problem with the HDMI/DP connection.  I found this thread on Debian forums describing a similar situation, although not identical to mine.  But to be fair, what appears to be the same issue manifests itself differently on 2 very similar Lenovo ThinkCentre Tiny computers, but with different APUs.
https://forums.debian.net/viewtopic.php?t=156021

IF there is a bug in lxqt-config-monitor, it seems to be related to this HDMI/DP combination.  I hope this is helpful for LXQT devs.  I've ordered a new HDMI/DP cable (no adaptors) to see if that resolves the issue.  if it does, perhaps the issue is the HDMI/DP adaptor?  If it does not, perhaps it is related to how the OS "sees" that connection.

I'll report back once I've tested the new cable.  If there's anything else I can provide to help LXQT devs in troubleshooting, just ask.  I'm a big fan of the LXQT desktop and would be happy to help out in maybe the only way I am qualified to do so!  LOL!  THANK YOU!

**Comment by JamesShriner** [comment]
okay...scrap the HDMI/DP converter theory.  I've learned there are DP ports and DP++ ports.  I verified my ports are DP++ and purchased an HDMI/DP cable...no adaptors required.  The end result is the same.

Hardware specs provided above in inxi -Fxz output.  Symptoms described in initial post can be summarized in that it appears the lxqt-config-monitor application is spawning multiple "zombie" clients during idle/sleep and using up all 256 xclient connections; thereby preventing any new applications from being launched.  Using previously open terminal, I can pkill lxqt-conf (from the list of lsof -U applications) and xclient connections are immediately reduced to approximately 40, and I'm back in business until the next idle/sleep cycle.  as long as i keep a terminal open, I can manually intervene and restore function to the desktop.  if i forget, the ONLY way out is a manual power-off from device...OUCH!

Here's a quick summary of my experiences:
Lenovo ThinkCentre m715q:  AMD Ryzen5 Pro 2400GE cpu with RavenRidge Radeon Vega/Mobile gpu that requires amdgpu graphics driver.  There are 3 graphics ports on this machine, and all are labled DP++ ports.  But closer inspection reveals that 1 of the ports is a different "color" behind the back plate, and I suspect has been swapped out.  So even though it's labled DP++, it MAY just be DP.  I was using this port for graphics, as it is the only port NOT inline with USB ports, so I assumed it was "primary".  I've since switched to one of the "stock" DP++ ports.  But it doesn't seem to matter, far as I can tell.

 AOC 27" 16:9 widescreen monitor:  capable of 1920x1080 resolution (which is the reason I bought this monitor 7+ years ago.  Rear ports include 2 HDMI and 1 recently discovered "recessed" VGA port. 

 

So possible connections with the Ryzen5 are (monitor port/computer port) VGA/DP and HDMI/DP.  In order of preference:

- HDMI/DP boots without issues, and LightDM greeter displays full 1920x1080 resolution, and login to LXQT DE displays and behaves perfectly while using.  When idle/sleep occurs, zombie spawns create runaway xclient connnections until 256 are reached, at which time NO NEW GUIs can be opened.  Already open GUIs (including terminal) work just fine; terminal will open new tabs in existing window, but no new windows of terminal will open.
- VGA/DP requires nomodeset to boot, which invokes the vesa driver, which behaves just fine; but no amdgpu graphics that I paid for. 

But at least the vesa driver behaves and seems stable during my experiments.  With the results above, I'm now connected to the monitor via the DP++ port, with DP/VGA adaptor, and plugged into the VGA port of the monitor.  (perhaps THAT is a clue?)  It's not ideal, but it's better than pkill-ing zombie clients after every idle/sleep session.  While I view this as an acceptable workaround, for the time being, I'd really prefer that amdgpu driver, but something isn't playing nice on my system.  And I'm at a loss as to what else to try.  Hoping SOMEONE on this forum has an idea for the long-term fix.

**Reply by JamesShriner** [reply]
I think I'm just talking to myself at this point, but in case anyone else experiences a similar issue, I now have this resolved.

The short version is this:  swapped out monitor hardware and cable and issue went away.  as a matter of thoroughness, replaced the new cable with cable used during monitor errors.  Again, no errors reported, so cable was eliminated as cause; and monitor was confirmed as issue.  I don't know what the issue could have been, as the AOC E2752S has been connected via HDMI to Lubuntu 16.04 with LXDE; and then Debian 10 with LXQT 0.14; with NO ISSUES whatsoever....EVER.  But for some reason, the combination of Debian 12, LXQT 1.2 and/or hardware connections created an issue.

I'm now using a 24" LG monitor with DP connection on monitor, and using DP cable to connect to DP++ port on machine.  no zombie xclients being spawned during idle/sleep.  Going 24 hours now without issues, so I think it's resolved.

Side note:  Debian 12 and LXQT 1.2 are SMOOTH like butter on this setup.  If early indications are reliable, this might be the best combination of OS and DE that i've ever experienced?!  \m/

**Comment by l3u** [comment]
Just for the record: I ran into (most probably) this exact issue: https://github.com/lxqt/lxqt/discussions/2507

Only difference is that I'm on Devuan Daedalus, but this is most probably the same as Debian concerning this. Has this been reported upstream? Is there some fix on the way?!

**Reply by JamesShriner** [reply]
I didn't get much help or direction on this, but I did get it resolved.  I read your issue and agree, sounds like the exact same issue.  If you read through my posts above, you'll see my troubleshooting efforts to isolate variables; which may or may not prove helpful to you.

Long story short...I firmly believe my issue was caused by a hardware incompatibility of some sort, related to DP+ port.  My computer has 3 DP ports, but that's it.  DP, and nothing else.  My monitor had HDMI and VGA inputs.  Using adaptors, the DP to VGA worked fine, but I had to boot with nomodeset parameter, which invoked the Mesa driver rather than the desired Radeon driver.  Using adaptors, the DP to HDMI booted fine, with no issues, but yielded the zombie xclient issues.  I even swapped out cables to eliminate a "bad cable" as a potential culprit.

My resolution occurred when I realized that my wife's new monitor had DP, HDMI, and VGA inputs.  I hooked up my machine to her monitor and booted up using each of those inputs, in succession (with nomodeset on the VGA input), with no zombie xclients over a several day period.  (my xclients would exceed connections in a short period....30 minutes(?) of sleep).  In my mind, that isolated variables to the Monitor hardware, cables, and/or cable adaptors.  So I purchased cable of DP to HDMI connection, with no adaptors involved (except DP to VGA, which required an adaptor).  But each result was duplicated, regardless of cable and/or adaptor.  That suggested to me that monitor was the issue, or at least the trigger, even though I had ZERO issues with it from date of purchase.

With that knowledge, I swapped monitors and have been using the DP cable connection from my machine to the monitor, with no issues whatsoever since.  The monitor that I was experiencing zombie xclients with had never shown a single problem in the prior 7-8 years of use, with HDMI to HDMI connection, and continues to be used on a different machine with HDMI to HDMI connection; with no problems whatsoever.  At this point, I can only conclude that the DP ports are central to the problem (for me, at least).

I hope this is helpful for you in your situation.  Are you connected to your monitor via DP?  If not, it's something else, obviously.  But I'm willing to bet that DP is somehow a factor in identifying the bug.  FWIW...

**Reply by l3u** [reply]
Thanks a lot for your reply and for sharing your knowledge!

I'm using a monitor I have been using for years now, connected via HDMI. I also used this one with my old server before (which ran LXQt on Gentoo) with no problems ever. Maybe, I'll try to swap some monitors I have, or even buy one with a Display Port (I don't currently have one). Would be nice if I could also fix it, like you.

I wanted to use a slim as possible DE, because most of the time, I only need it to run a QEMU virtual machine. Well, after all, a possible solution would be to use KDE … but I run LXQt on some old notebooks, and – until now – it always worked.

What really puzzles me is: What the f**k is the real root cause of this?! This is really a very, very strange issue …

**Reply by JamesShriner** [reply]
I've been using LXQT for quite awhile now.  Started with Siduction LXQT, but then decided to go fixed release, so installed LXQT (0.12?) on Lubuntu (before Lubuntu switched), but have been using LXQT on Debian 10 Buster since Buster came out.  All configurations were with that HDMI monitor, using an HDMI to HDMI connection.

I upgraded Debian 10 to Debian 12 (fresh install) on my server installation, also using an HDMI to HDMI connection, with an AMD APU processor.  No issues whatsoever.

I installed Debian 12 (fresh install) on a refurbed Lenovo m715q, as my new workhorse, but was forced to change the HDMI to HDMI connection to a DP to HDMI connection, as the Lenovo only has DP ports.  That's where ALL my problems have occurred, as outlined above.  But like I said, resolved once I switched out monitors to a DP to DP connection.

At this point, I can only conclude 1 of 2 things:

- DP to HDMI connnections are a problem in specialized configurations with Debian/LXQT, or
- that older monitor is not forward compatible with DP connections, despite DP to HDMI adaptor

I would like to know the root cause also.  I've surfed google hits and found very few instances of similar situations, with very little information regarding solutions.  I'd like to know the underlying root cause also.  Is it as simple as "hardware incompatibility"?  And if so, isn't a software patch a potential solution?  Someone smarter than me will have to show the way.

Until then, this is all I have for you.  But I can 100% confirm that it resolved my problem!  No theoretical posturing here.  But if you need to bounce something off me, I'll do the best I can for you.  Good luck!

**Reply by l3u** [reply]
As far as I can see, a DP-HDMI connection would not cause this, as DP was never used here.

Anyway, I'll try to switch monitors, and possibly even to use a DP-DP connection to a yet-to-buy monitor.

Possibly, only few people are affected by this, and a dev never saw it …

**Reply by tsujan** [reply]
My wild guess is that the issue is caused by a faulty package in Debian Bookworm.

**Reply by JamesShriner** [reply]
well in that case, I'd say you have completely destroyed my DP+ theory.  Still progress though, right?

What are you using for APU/GPU?  My problem machine has a Ryzen 5 APU, with the seemingly problematic Raven Ridge/Radeon Vega chipset.  But it seems that was addressed back in the 5.15(?) kernel, and I'm running a 6.1.0-13 kernel.  Graphics driver is amdgpu.  I could invoke Mesa driver with nomodeset boot parameter, which ran without issue, but didn't really fully utilize amdgpu capabilities.

Another variable I was considering, after the fact, is that my current install uses LightDM as session manager; while SDDM was manager in prior LXQT installs.  I'm wondering if LightDM could be a potential culprit?

**Reply by JamesShriner** [reply]
> As far as I can see, a DP-HDMI connection would not cause this, as DP was never used here.  Anyway, I'll try to switch monitors, and possibly even to use a DP-DP connection to a yet-to-buy monitor.  **Possibly, only few people are affected by this**, and a dev never saw it …

Best I can tell, there's about 5 of us, based on google searches.  LOL!

I zero'd in on LXQT, because I tried other distros and desktops in a live session, and could not duplicate the issue.  Not with other distros.  Not with other desktops in the Debian family.  So I concluded it was likely an LXQT issue, rather than Debian.  But maybe it's a Debian LXQT issue?  OTOH, I can't believe there aren't more of us, if that's the case.  You using Devuan is an interesting clue, as it seems to support tsujan's guess of a mis-configured Bookworm package, which is showing up "downstream" in Devuan.  Good luck waiting for the fix for 5 affected users, LOL.

**Reply by l3u** [reply]
I run an Intel Core i5, using the integrated graphics card. And I use SDDM.

I'd really love to track this down …

**Reply by tsujan** [reply]
> Good luck waiting for the fix for 5 affected users, LOL.

That's why I asked [elsewhere](https://github.com/lxqt/lxqt/discussions/2507#discussioncomment-7842821) if there's a reliable repository with backported packages of a recent version of LXQt for Debian Bookworm.

**Reply by l3u** [reply]
On Gentoo, doing something like that was so easy … ;-) I'm personally a Debian/Devuan n00b, so I can't tell how to test recent versions or where to get them. Building them locally wouldn't be so easy, would it?

Well, what I'll definitely try is to simply change nothing on my setup and use KDE to see if it works, or if I get some strange zombie processes after idling there, too. If KDE works, I think we can at least narrow it down to some LXQt-specific package.

Even after thinking it through a lot, I have no idea how to debug this …

**Reply by tsujan** [reply]
> Building them locally wouldn't be so easy, would it?

Not if you're new to Debian. It requires knowing how to make deb packages. It'll be easy once you gather some experience about Debian. However, we're talking about a whole DE here; that needs much more time and experience.

Anyway, I recommend you don't install anything without deb packages.

**Reply by JamesShriner** [reply]
l3u.....as an experienced Gentoo user, no doubt you possess the skillset to build from source.  But even so, you probably don't want to try PPAs or even to build your own from source.  At least not in Debian.  Your system, so your call, but Debian forums are littered with users warning to not create a "franken-debian".  If you use packages outside of Debian, or outside of the Debian version you're using, there's a good chance you'll be on your own with any potential issues.  But that's your decision to make...

appreciate you sharing that you have an Intel Core i5 apu, and using SDDM as session manager.  I think that eliminates the possibility that any bug is related to Ryzen5, or with LightDM.  Cross them off the list of potential suspects.  As the list of potential suspects continues to dwindle, it is looking more and more likely that tsujan's wild guess is the most likely solution....a bad LXQT package in Debian Bookworm.  Maybe lxqt-config-monitor?  or more likely, whatever process CALLS lxqt-config-monitor.  I am also at a loss as to where to look next.  Like I said....I solved MY problem.  But obviously, it's still there for other users.  Maybe my solution works for you, maybe it doesn't?

FWIW, I have used other distros in LiveSessions, on the exact same hardware while troubleshooting, with no issues.  I have used other versions of Debian in LiveSessions, with no issues.  I could only reproduce the error with the LXQT version of Debian 12.  And with the assumed number of Debian Bookworm LXQT installs, 5 potential users being affected (estimated number of google hits that are reasonably similar errors) just seems statistically improbable.  It really is puzzling.

**Reply by l3u** [reply]
I don't even think that lxqt-config-monitor itself is the problem. I thought it was, and thus, I moved it and created a script called lxqt-config-monitor that does nothing than logging that something just wanted to start lxqt-config-monitor. Then, this script became a zombie process …

This is really a very, very odd issue.

**Reply by l3u** [reply]
I have been a Gentoo user for almost 20 years now. I have to admit that I can't remember such a nasty bugger, without any fix or at least workaround in sight :-(

**Reply by JamesShriner** [reply]
your script idea is interesting.  seems to suggest that lxqt-config-monitor isn't the problem, but rather the mechanism....whatever is CALLING lxqt-config-monitor seems to be the root cause here.  You can increase your number of xclient connections from 256 to 512 or 1024, i think all the way to 2048; but all that will accomplish is "increasing your headroom".  Once you reach that limit (256 for me), no new xclient connections will be spawned, until you free up those zombie connections.

IDEA for a workaround!  Can you modify your script to pkill the lxqt-conf "process" referenced from the CLI "xrestop" in one of my linked threads?  My thought is that some unknown process is calling lxqt-config-monitor, and repeats until 256 is reached, at which time, the GUI freezes.  You replaced that process with a script, so the same thing occurs, but your script is the new zombie.....a "suicidal zombie" that kills itself every time it is invoked?  That's the theory, at least.

If you add a function in your script to pkill lxqt-conf (the process that xrestop identifies 256 times), it seems to me that your script would kill the zombie xclient each time it is called, rather than accumulate.  It's not a solution, but perhaps a simple workaround to buy you all the time you need until a real solution is identified?  Thoughts?

**Reply by l3u** [reply]
Well, that could probably mitigate the real issue, but that would be one nasty workaround ;-)

**Reply by l3u** [reply]
Whatever causes this has to be fixed upstream. Definitely.

**Reply by JamesShriner** [reply]
in theory, that would work fantastic.  one zombie xclient process spawned (your script), which then proceeds to pkill the very zombie xclient spawned.  If that works, you could be good forever, LOL!  If you decide to give that a try, let me know if it actually works?

I agree on the upstream fix, though.  For the first time, I no longer believe it's an LXQT issue.  It's something else, that's hijacking LXQT process and making LXQT appear to be the guilty party.  I guess that's progress, eh?!

**Comment by l3u** [comment]
Some additional information:

Simply switching the monitor off (physically) spawns an lxqt-config-monitor zombie process:

    $ ps -ef | grep lxqt-config-monitor
    serverh+ 26115  3339  0 09:12 ?        00:00:00 /usr/bin/lxqt-config-monitor -l
    serverh+ 26121 26106  0 09:12 pts/1    00:00:00 grep lxqt-config-monitor

And here's who does it:

    $ ps -fp 3339
    UID        PID  PPID  C STIME TTY          TIME CMD
    serverh+  3339  3333  0 Nov22 ?        00:25:52 lxqt-session

**Comment by l3u** [comment]
… and some more:

As there is no fix for this at the moment, I installed KDE for now.

Using this and turning off the monitor using the power management, the monitor is shortly turned off, but is turned on again immediately. Disabling the automatic source selection in the monitor's setting (the hardware settings inside the monitor, not something on my machine!) fixed this: Now, the monitor goes to sleep as it should.

Maybe, this is related?! I can't play around with this, as this is a production server in my practice, and I would have to shutdown my real Windows server I run via QEMU to launch it again using LXQt – and that's something I only want to do three or four times a year :-(

**Comment by tsujan** [comment]
I added "bug" label to https://github.com/lxqt/lxqt/issues/2509. Although rare, this should be a new bug in LXQt — otherwise, there wouldn't be three reports.

If the real cause is found and fixed, we should make a point release for the component that exhausts file handles under special circumstances. The problem is that no dev has succeeded in reproducing it. In such cases, there may be very difficult to find the needle in the haystack (there was one [case](https://github.com/lxqt/qps/issues/447) similar to this, where the "needle" was found at last, thanks to reporters' cooperation).
